src/v/raft/follower_stats.h

```diff
@@ -54,7 +54,8 @@ class follower_stats {
     }
 
     iterator emplace(vnode n, follower_index_metadata m) {
-        auto [it, success] = _followers.insert_or_assign(n, std::move(m));
+        _followers.erase(n);
+        auto [it, success] = _followers.emplace(n, std::move(m));
```

Comment:
that's unfortunate. the only way around this that comes to mind is to implement the move assignment operator for `follower_index_metadata` using placement-new and delegate to the move constructor. dunno if that is worth it.